starknet_py/net/full_node_client.py

```diff
@@ -849,74 +871,100 @@ def _create_broadcasted_declare_v2_properties(transaction: DeclareV2) -> dict:
+
+    return {
+        **_create_broadcasted_txn_v3_common_properties(transaction),
+        **declare_v3_properties,
+    }
+
+
+def _create_broadcasted_invoke_properties(transaction: Union[Invoke, InvokeV3]) -> dict:
     invoke_properties = {
         "sender_address": _to_rpc_felt(transaction.sender_address),
         "calldata": [_to_rpc_felt(data) for data in transaction.calldata],
     }
+
+    if isinstance(transaction, InvokeV3):
+        return {
+            **_create_broadcasted_txn_v3_common_properties(transaction),
+            **invoke_properties,
+            "account_deployment_data": [
+                _to_rpc_felt(data) for data in transaction.account_deployment_data
+            ],
+        }
+
     return invoke_properties
 
 
-def _create_broadcasted_deploy_account_properties(transaction: DeployAccount) -> dict:
+def _create_broadcasted_deploy_account_properties(
+    transaction: Union[DeployAccount, DeployAccountV3]
+) -> dict:
     deploy_account_txn_properties = {
         "contract_address_salt": _to_rpc_felt(transaction.contract_address_salt),
         "constructor_calldata": [
             _to_rpc_felt(data) for data in transaction.constructor_calldata
         ],
         "class_hash": _to_rpc_felt(transaction.class_hash),
     }
+
+    if isinstance(transaction, DeployAccountV3):
+        return {
+            **_create_broadcasted_txn_v3_common_properties(transaction),
+            **deploy_account_txn_properties,
+        }
+
     return deploy_account_txn_properties
 
 
 def _create_broadcasted_txn_common_properties(transaction: AccountTransaction) -> dict:
     broadcasted_txn_common_properties = {
         "type": transaction.type.name,
-        "max_fee": _to_rpc_felt(transaction.max_fee),
         "version": _to_rpc_felt(transaction.version),
         "signature": [_to_rpc_felt(sig) for sig in transaction.signature],
         "nonce": _to_rpc_felt(transaction.nonce),
     }
-    return broadcasted_txn_common_properties
-
-
-def _to_storage_key(key: int) -> str:
-    """
-    Convert a value to RPC storage key matching a ``^0x0[0-7]{1}[a-fA-F0-9]{0,62}$`` pattern.
-
-    :param key: The key to convert.
-    :return: RPC storage key representation of the key.
-    """
 
-    hashed_key = hex(key)[2:]
-
-    if hashed_key[0] not in ("0", "1", "2", "3", "4", "5", "6", "7"):
-        hashed_key = "0" + hashed_key
-
-    hashed_key = "0x0" + hashed_key
-
-    if not re.match(r"^0x0[0-7]{1}[a-fA-F0-9]{0,62}$", hashed_key):
-        raise ValueError(f"Value {key} cannot be represented as RPC storage key.")
-
-    return hashed_key
-
-
-def _to_rpc_felt(value: Hash) -> str:
-    """
-    Convert the value to RPC felt matching a ``^0x(0|[a-fA-F1-9]{1}[a-fA-F0-9]{0,62})$`` pattern.
+    if _extract_tx_version(transaction.version) < 3 and hasattr(transaction, "max_fee"):
+        broadcasted_txn_common_properties["max_fee"] = _to_rpc_felt(
+            transaction.max_fee  # pyright: ignore
+        )
```

Comment:
Not necessarily in this PR, but I think it could and should be refactored, as the logic seems very unintuitive to me.
1. This should apply all common properties, but in reality it applies only intersection of v1 and v3 common properties + if needed the entirety of v1 common properties.
2. At the same time, all v3 common properties are actually applied by type-specific functions.
3. I think logic in `_create_broadcasted_txn` should be refactored:
```python
    txn_map = {
        TransactionType.DECLARE: _create_broadcasted_declare_properties,
        TransactionType.INVOKE: _create_broadcasted_invoke_properties,
        TransactionType.DEPLOY_ACCOUNT: _create_broadcasted_deploy_account_properties,
    }

    common_properties = _create_broadcasted_txn_common_properties(transaction)
    transaction_specific_properties = txn_map[transaction.type](transaction)
```
First, we should create all common properties by version, regardless of the type.
Then, we should apply the rest of the properties based both on type and version.

Reply:
Totally agree, thank you for your comment! 

I believe we can also leverage `TypesOfTransactionsSchema().dump(transaction)` here to get broadcasted dict. Only `Declare` may require some additional work. I've created an issue so it won't get lost: https://github.com/software-mansion/starknet.py/issues/1246